docs/requirements.txt

```diff
@@ -0,0 +1,12 @@
+# Sphinx-related requirements.
+sphinx
+sphinx-book-theme>=1.0.1
+myst-nb
+myst-parser[linkify]
+sphinx-book-theme
+sphinx-copybutton
+sphinx-design
+# Avoiding an issue with the collapsible sidebar.
+pydata-sphinx-theme<0.16.0
+# To generate API documentation.
+sphinx-autoapi
```

Comment:
I also had to 
```suggestion
sphinx-autoapi
sphinx-autodoc2
```

Reply:
I accidentally pushed some experiments with `autodoc2`, but the generation was not working (as you noted above despite trying various configurations of `paths/module/exclude` so will stick to `autoapi`